R/data-combined.R

```diff
@@ -108,24 +109,23 @@ DataCombined <- R6::R6Class(
         names <- ifelse(is.na(names), datasetNames, names)
       }
 
-      private$.verifyExistingNames(newNames = c(datasetNames, names))
-
       # Update private fields and bindings for the new setter call
-
       private$.dataCombined <- private$.updateDataFrame(
         private$.dataCombined,
-        private$.dataSetToDataFrame(dataSets, names)
+        private$.dataSetToDataFrame(dataSets, names),
+        silent = silent
       )
 
-
-      private$.extractBindings()
-
+      names <- names %||% datasetNames
+      # Reset data transformations
       self$setDataTransformations(names)
 
-      if (!is.null(groups)) {
-        self$setGroups(names %||% datasetNames, groups)
-      }
+      private$.updateGroups(names, groups)
 
+      # Set data type
+      for (name in names) {
+        private$.dataType[[name]] <- "observed"
```

Comment:
don't we have an enum for data types?

Reply:
No, and at the moment I wouldn't know where the user coud use the enum.